### PR TITLE
Prepare release notes for 2.4-M2

### PR DESCRIPTION
**Pull request by leonard84**

### CODE REVIEW

**Comment by leonard84** [comment]
I've excluded all bot commits and purely internal changes

File: docs/release_notes.adoc [hunk 1]
```diff
@@ -13,40 +15,73 @@ include::include.adoc[]
...
+** by adding missing closure hints for Spy(T, Closure) spockPull:1786[]
+** by adding GDSL/DSLD for ConditionalExtension's annotations spockPull:1808[]
+** by making `spock.gdsl` fail-safe and cover some more cases spockPull:1783[]
+* Improve RunContexts behavior in multithreaded executions spockPull:1758[], spockPull:1846[]
+* Improve generic handling by replacing `gentyref` code with https://github.com/leangen/geantyref[geantyref] library spockPull:1743[]
+* Improve handling of passing `null` to `thrown()` to be consistent with mock object creation spockPull:1799[]
+* Improve `old(...)` calls by validating their argument count spockPull:1810[]
+* Improve robustness of the AST transformation against missing classes in the compile classpath spockPull:1704[]
+* Improve handling of `shared.` conditions in `@IgnoreIf`/`@Requires`, the condition is now checked before creating data providers spockPull:1711[]
+** This is now a required dependency used by spock: `io.leangen.geantyref:geantyref:1.3.14`
```
Comment (AndreasTu) [inline docs/release_notes.adoc:61]: The line 61 `This is now a required` should be moved to line 57 after the `Improve generic handling by replacing`

File: docs/release_notes.adoc [hunk 2]
```diff
@@ -13,40 +15,73 @@ include::include.adoc[]
...
 * Fix exception when configured `baseDir` was not existing, now `@TempDir` will create the baseDir directory if it is missing.
 * Fix bad error message for collection conditions, when one of the operands is `null`
+* Fix docs about initializer method interceptors spockPull:1666[]
 * Fix possible deadlock, when blocking in mock response generators spockPull:1885[]
+* Fix SpringSpy not working with `DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD` spockPull:1869[]
+* Fix null handling for collection conditions spockPull:1858[]
+* Fix interceptor contexts spockPull:1676[]
+* Fix properly call `IRunListener.specSkipped` and `.featureSkipped` spockPull:1811[]
+* Fix defining responses for additionalInterfaces spockPull:1730[]
+* Fix lost execution mode for iteration nodes of implied data-driven features spockPull:1615[]
```
Comment (AndreasTu) [inline docs/release_notes.adoc:76]: Line 77 `This fixes the issues`  should be moved to line 71 below `Fix possible deadlock,`

**Review by AndreasTu — CHANGES_REQUESTED** [review]

File: docs/release_notes.adoc [hunk 3]
```diff
@@ -5,6 +5,8 @@ include::include.adoc[]
 
 == 2.4 (tbd)
```
Comment (AndreasTu) [inline docs/release_notes.adoc:6]:
```suggestion
== 2.4-M2
```

**Review by AndreasTu — CHANGES_REQUESTED** [review]

File: docs/release_notes.adoc (hunk 3 above)
Comment (leonard84) [reply docs/release_notes.adoc:6]: The (tbd) is the placeholder for the date

File: docs/release_notes.adoc (hunk 1 above)
Comment (leonard84) [reply docs/release_notes.adoc]: good catch

File: docs/release_notes.adoc (hunk 2 above)
Comment (leonard84) [reply docs/release_notes.adoc:77]: good catch

**Pull request merged**